ui/BUILD.bazel

```diff
@@ -0,0 +1,22 @@
+load("//bazel/ui:react.bzl", "react_project")
+
+exports_files([
+    "package.json",
+    "merge.js",
+])
+
+# Note this patch (May 25th,
+filegroup(
+    name = "git-patches",
+    srcs = glob(["patches/*"]),
+    visibility = ["//visibility:public"],
+)
+
+react_project(
+    name = "ptunnel",
+    srcs = "//ui/ptunnel:srcs",
```

Comment:
for `srcs`, `public`, `git-patches` and .. `ui-extras`, I would recommend not hard coding the copy file, rather have something like:

```
react_project(
  name = "ptunnel",
  elements = {
    "patches": [":git-patches"],
    "public": ["//ui/ptunnel:public"],
    "src": ["//ui/putnnel:srcs"],
  }
)
```

so in the macro implementation you can use a for loop, while allowing arbitrary paths to be brought into the rule?
Or you may just want to structure the code this way in the macro: first build the dict, and then do a loop.

Reply:
I agree with making it self contained, but I disagree with the dictionary. I would prefer for named inputs, considering there's no way to run a modern project without them, and for easy of editor use. For optional configs sure, but I view the current inputs as required.